public/app/plugins/panel/logs/LogsPanel.tsx

```diff
@@ -52,6 +60,25 @@ export const LogsPanel = ({
   const logsContainerRef = useRef<HTMLDivElement>(null);
   const [contextRow, setContextRow] = useState<LogRowModel | null>(null);
   const [closeCallback, setCloseCallback] = useState<(() => void) | null>(null);
+  const timeRange = data.timeRange;
+  const urlParams = urlUtil.getUrlSearchParams();
+  const panelStateEncoded = urlParams?.panelState;
+  const getLogsPanelState = (): LogsPermalinkUrlState | undefined => {
+    if (
```

Comment:
This component is growing really quickly with the things we added recently, such as context, infinite scrolling (soon), and more. Would you mind moving this to a function somewhere? It can be defined in this same file.